src/server/api/lib/message-sending.js

```diff
@@ -392,23 +392,32 @@ export async function saveNewIncomingMessage(messageInstance) {
   let updateQuery = r.knex("campaign_contact").limit(1);
 
   if (OPT_OUT_TRIGGERS.includes(cleanedUpText)) {
-    updateQuery = updateQuery.update({ message_status: "closed" });
-
     const { id: organizationId } = await r
       .knex("organization")
       .first("organization.id")
       .join("campaign", "organization_id", "=", "organization.id")
       .join("assignment", "campaign_id", "=", "campaign.id")
       .where({ "assignment.id": assignment_id });
 
-    const optOutId = await cacheableData.optOut.save(r.knex, {
-      cell: contact_number,
-      reason: "Automatic OptOut",
-      assignmentId: assignment_id,
-      organizationId
-    });
+    const existingOptOutQuery = r
+      .knex("opt_out")
+      .first("id")
+      .where({ cell: contact_number });
+
+    const existingOptOut = await existingOptOutQuery;
+
+    if (existingOptOut === undefined) {
+      updateQuery = updateQuery.update({ message_status: "closed" });
+
+      const optOutId = await cacheableData.optOut.save(r.knex, {
+        cell: contact_number,
+        reason: "Automatic OptOut",
+        assignmentId: assignment_id,
+        organizationId
+      });
 
-    await queueExternalSyncForAction(ActionType.OptOut, optOutId);
+      await queueExternalSyncForAction(ActionType.OptOut, optOutId);
+    }
```

Comment:
1. For closing convos - I think we should change the behavior to close the convo when we get _any_ inbound message where the contact `is_opted_out` The current behavior is a little weird because if someone who's opted out responds with a regular message, we just leave that convo in `needsResponse`, but if they respond with a stop word we close the convo. 

2. External Sync - I'm probably just not familiar enough with how this is supposed to work! Can you say more about how sending over the repeated event is helpful? My thinking was that we want to notify the external system about when the contact getting opted out, not how many times they repeated that.

Reply:
Both fair points! I think my only concern is being consistent with what happens "normally" (when texters are handling replies) with what auto handling is doing here. Since texters don't see opted out convos:

1. Texters currently can't opt someone back in, only a good sweeping team can from message review. @politics-rewired/organizing what message status (needsResponse/closed) do we think is most useful for sweepers for:
&nbsp; a. Stop words
&nbsp; b. Regular responses
do we need to be consistent?
2. We don't have a way to mark repeated opt out requests as a texter. I guess a sweeper could do it from message review but it's not so intuitive.

If we think that auto handling doing different things from texters isn't a problem then I think you're right on both questions

Reply:
Yeah, repeated events should be fine! If we have repeated events happening, they should be passed on.